tests/test_typedefs.py

```diff
@@ -0,0 +1,12 @@
+"""Test aiohttp.typedefs."""
+from typing import TYPE_CHECKING, Awaitable, Callable
+
+from aiohttp.typedefs import Handler, Middleware
+
+if TYPE_CHECKING:
+    from aiohttp.web import Request, StreamResponse
+
+
+def test_middleware() -> None:
+    """Test aiohttp.typedefs.Middleware."""
+    assert Middleware == Callable[["Request", Handler], Awaitable["StreamResponse"]]
```

Comment:
I'm not sure this is really testing anything.

I think a better option would be to add the return types throughout test_web_middlewares.py and then this function can be annotated with the new `Middleware` type as well:
https://github.com/aio-libs/aiohttp/blob/master/tests/test_web_middleware.py#L62

Any problems with the type in future should then be caught by Mypy from this usage.